src/main/java/com/cburch/logisim/std/memory/AbstractFlipFlopHdlGeneratorFactory.java

```diff
@@ -60,73 +60,76 @@ public LineBuffer getModuleFunctionality(Netlist nets, AttributeSet attrs) {
         .pair("invertClock", INVERT_CLOCK_STRING)
         .pair("Clock", HdlPorts.CLOCK)
         .pair("Tick", HdlPorts.TICK)
+        .empty()
         .addRemarkBlock("Here the output signals are defined")
         .add("""
-             {{assign}}Q       {{=}}s_current_state_reg;
-             {{assign}}Q_bar   {{=}}{{not}}(s_current_state_reg);
-             """)
-        .add(Hdl.isVhdl()
-            ? "s_clock {{=}} {{Clock}} WHEN {{invertClock}} = 0 ELSE NOT({{Clock}});"
-            : "assign s_clock {{=}} ({{invertClock}} == 0) ? {{Clock}} : ~{{Clock}};")
-        .addRemarkBlock("Here the update logic is defined")
-        .add(getUpdateLogic())
-        .add("");
-    if (Hdl.isVerilog()) {
+             {{assign}}q       {{=}}s_currentState;
+             {{assign}}qBar    {{=}}{{not}}(s_currentState);
+             """);
+    if (Hdl.isVhdl()) {
+      contents.addVhdlKeywords()
+          .add("s_clock {{=}}{{Clock}} {{when}} {{invertClock}} = 0 {{else}} {{not}}({{Clock}});")
+          .empty();
+    } else {
       contents
+          .add("assign s_clock {{=}}({{invertClock}} == 0) ? {{Clock}} : ~{{Clock}};")
+          .empty()
           .addRemarkBlock("Here the initial register value is defined; for simulation only")
           .add("""
                initial
                begin
-                  s_current_state_reg = 0;
+                  s_currentState = 0;
                end
-
-               """);
+               """)
+          .empty();
```

Comment:
Why not adding empty line to the text block directly?

Reply:
This has to do with the .split("\n") operation, it will "remove" the last empty line, hence it is a "work around".